### PR TITLE
vmalert: enable configuring explicit path

### PR DESCRIPTION
**Pull request by alxric**
Currently, whatever you configure as `remoteWrite.url` , the path `/api/v1/write` will always be appended to it.
This causes problems when your remote host for one reason or the other does not serve that path, so this PR will add the option to override it with an absolute path if so desired, otherwise keep using the current behaviour.

### CODE REVIEW

**Comment by f41gh7** [comment]
Hm, probably, its better to introduce flag -  `-remoteWrite.disablePathAppend`, which would control path concatenation for remoteWrite?
  And with `./bin/vmalert -remoteWrite.url=http://some-url:8428/api/v2/write -remoteWrite.disablePathAppend` it would be possible to use absolute url for remoteWrite.

@hagen1778  WDYT?

**Comment by hagen1778** [comment]
I agree with @f41gh7 - `-remoteWrite.disablePathAppend` will simplify the change while allowing the same functionality. @alxric, do you have any objections?

**Comment by alxric** [comment]
> I agree with @f41gh7 - `-remoteWrite.disablePathAppend` will simplify the change while allowing the same functionality. @alxric, do you have any objections?

Nope, none at all. I agree, that is a cleaner approach. Pushed suggested changes

File: app/vmalert/README.md [hunk 1]
```diff
@@ -502,6 +502,8 @@ The shortlist of configuration flags is the following:
     	Optional TLS server name to use for connections to -remoteWrite.url. By default the server name from -remoteWrite.url is used
   -remoteWrite.url string
     	Optional URL to VictoriaMetrics or vminsert where to persist alerts state and recording rules results in form of timeseries. E.g. http://127.0.0.1:8428
+  -remoteWrite.disablePathAppend
+    	Optionally disable appending '/api/v1/write' to the remoteWrite url, enabling configuring an absolute path
```
Comment (hagen1778) [inline app/vmalert/README.md:506]:
```suggestion
     Whether to disable automatic appending of '/api/v1/write' path to the configured -remoteWrite.url.
```

File: app/vmalert/remotewrite/init.go [hunk 2]
```diff
@@ -27,6 +27,8 @@ var (
 		"By default system CA is used")
 	tlsServerName = flag.String("remoteWrite.tlsServerName", "", "Optional TLS server name to use for connections to -remoteWrite.url. "+
 		"By default the server name from -remoteWrite.url is used")
+	disablePathAppend = flag.Bool("remoteWrite.disablePathAppend", false, "Optionally disable appending '/api/v1/write' to the remoteWrite url, "+
```
Comment (hagen1778) [inline app/vmalert/remotewrite/init.go:30]:
```suggestion
	disablePathAppend = flag.Bool("remoteWrite.disablePathAppend", false, "Whether to disable automatic appending of '/api/v1/write' path to the configured -remoteWrite.url.")
```

File: app/vmalert/remotewrite/remotewrite.go [hunk 3]
```diff
@@ -56,6 +57,8 @@ type Config struct {
 	WriteTimeout time.Duration
 	// Transport will be used by the underlying http.Client
 	Transport *http.Transport
+	// DisablePathAppend can be used to not automatically append '/api/v1/write' to the remote write url
```
Comment (hagen1778) [inline app/vmalert/remotewrite/remotewrite.go:60]:
```suggestion
	// DisablePathAppend prevents appending of '/api/v1/write' path to the addr
```

File: docs/vmalert.md [hunk 4]
```diff
@@ -506,6 +506,8 @@ The shortlist of configuration flags is the following:
     	Optional TLS server name to use for connections to -remoteWrite.url. By default the server name from -remoteWrite.url is used
   -remoteWrite.url string
     	Optional URL to VictoriaMetrics or vminsert where to persist alerts state and recording rules results in form of timeseries. E.g. http://127.0.0.1:8428
+  -remoteWrite.disablePathAppend
+    	Optionally disable appending '/api/v1/write' to the remoteWrite url, enabling configuring an absolute path
```
Comment (hagen1778) [inline docs/vmalert.md:510]:
```suggestion
    	Whether to disable automatic appending of '/api/v1/write' path to the configured -remoteWrite.url.
```

**Review by hagen1778 — CHANGES_REQUESTED** [review]
Thanks! Small changes

**Comment by hagen1778** [comment]
Build fails. Looks like in suggestion I broke the line format.

**Comment by alxric** [comment]
> Build fails. Looks like in suggestion I broke the line format.

Ah, missed to verify that I could build locally after changes. Should be fine now

**Comment by hagen1778** [comment]
Thanks for contribution!

**Comment by valyala** [comment]
FYI, the pull request has been included in [vmalert v1.64.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Pull request merged**